services/chronikService.ts

```diff
@@ -157,9 +162,67 @@ export class ChronikBlockchainClient implements BlockchainClient {
     return details
   }
 
+  private getWsConfig (): WsConfig {
+    return {
+      onConnect: (e: ws.Event) => { console.log(`Chronik WebSocket connected, message type: ${e.type}`) },
+      onMessage: (msg: SubscribeMsg) => { void this.processWsMessage(msg) },
+      onError: (e: ws.ErrorEvent) => { console.log(`WebSocket error, message type: ${e.type} | message: ${e.message} | error: ${e.error as string}`) },
+      onEnd: (e: ws.Event) => { console.log(`WebSocket ended, message type: ${e.type}`) },
+      autoReconnect: true
+    }
+  }
+
+  private async processWsMessage (msg: SubscribeMsg): Promise<void> {
+    // delete unconfirmed transaction from our database
+    if (msg.type === 'RemovedFromMempool' || msg.type === 'Confirmed') {
+      const transactionsToDelete = await fetchUnconfirmedTransactions(msg.txid)
+      await deleteTransactions(transactionsToDelete)
+    }
+    // create unconfirmed or confirmed transaction
+    if (msg.type === 'AddedToMempool' || msg.type === 'Confirmed') {
+      const transaction = await await this.chronik.tx(msg.txid)
+      const addressWithTransactions = await this.getPrismaTransactionsForSubscribedAddresses(transaction)
```

Comment:
I am confused about this method which is being called. It returns a dictionary where the keys are all addresses subscribed, why? This transaction will probably have to do with only 1 or 2 of the subscribed addresses AFAIK, so why go through all subscribed addresses and then create the transaction for all of them? Looks wrong to me unless I am not understanding something correctly

Reply:
I am probably not understanding something correctly for I have tested and seems to be working.

Reply:
A transaction can have multiple inputs and outputs, we don't know which of the subscribed addresses are in the transaction, so we get the amount transacted to each one of them in this transaction and then filter the result by only those addresses that have any amount transacted before returning:

`return addressWithTransactions.filter(
   addressWithTransaction => addressWithTransaction.transaction.amount > new Prisma.Decimal(0)
 )`

Reply:
got it, very cool